versioned/persist/tests/src/main/java/org/projectnessie/versioned/persist/tests/extension/NessieMockedTracingExtension.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.persist.tests.extension;
+
+import static org.junit.platform.commons.util.AnnotationUtils.findAnnotatedFields;
+import static org.junit.platform.commons.util.ReflectionUtils.makeAccessible;
+
+import io.opentracing.mock.MockTracer;
+import io.opentracing.util.GlobalTracer;
+import java.lang.reflect.Modifier;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionConfigurationException;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.junit.platform.commons.util.ExceptionUtils;
+
+/**
+ * Extension that allows injecting a {@link MockTracer} on {@link NessieDbTracer} annotated fields
+ * and parameters.
+ *
+ * <p>Note that the {@link MockTracer} instance is constant per class - like a {@code static final}
+ * field.
+ */
+public class NessieMockedTracingExtension
+    implements BeforeAllCallback, BeforeEachCallback, ParameterResolver {
+  private static final Namespace NAMESPACE = Namespace.create(NessieMockedTracingExtension.class);
+
+  private static final MockTracer TRACER = new MockTracer();
```

Comment:
Could we use `MockTracer` to replace our `TestTracer`?

Reply:
Probably - but out of scope for this PR IMO.
`TestTracer` works fine - and since opentracing is rather deprecated, we'll have to touch it when moving to opentelemetry anyway.

Reply:
fair enough.